presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergTableProperties.java

```diff
@@ -24,49 +30,96 @@
+import static com.facebook.presto.iceberg.IcebergWarningCode.USE_OF_DEPRECATED_TABLE_PROPERTY;
 import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
 import static com.facebook.presto.spi.session.PropertyMetadata.integerProperty;
 import static com.facebook.presto.spi.session.PropertyMetadata.longProperty;
 import static com.facebook.presto.spi.session.PropertyMetadata.stringProperty;
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.lang.String.format;
 import static java.util.Locale.ENGLISH;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
 import static org.apache.iceberg.TableProperties.UPDATE_MODE;
 
 public class IcebergTableProperties
 {
+    /**
+     * Please use  {@link TableProperties#DEFAULT_FILE_FORMAT}
+     */
+    @Deprecated
     public static final String FILE_FORMAT_PROPERTY = "format";
+    private static final String DEFAULT_FORMAT_VERSION = "2";
     public static final String PARTITIONING_PROPERTY = "partitioning";
-
     public static final String SORTED_BY_PROPERTY = "sorted_by";
     public static final String LOCATION_PROPERTY = "location";
+    /**
+     * Please use  {@link TableProperties#FORMAT_VERSION}
+     */
+    @Deprecated
     public static final String FORMAT_VERSION = "format_version";
+    /**
+     * Please use  {@link TableProperties#COMMIT_NUM_RETRIES}
+     */
+    @Deprecated
     public static final String COMMIT_RETRIES = "commit_retries";
+    /**
+     * Please use  {@link TableProperties#DELETE_MODE}
+     */
+    @Deprecated
     public static final String DELETE_MODE = "delete_mode";
+    /**
+     * Please use  {@link TableProperties#METADATA_PREVIOUS_VERSIONS_MAX}
+     */
+    @Deprecated
     public static final String METADATA_PREVIOUS_VERSIONS_MAX = "metadata_previous_versions_max";
+    /**
+     * Please use  {@link TableProperties#METADATA_DELETE_AFTER_COMMIT_ENABLED}
+     */
+    @Deprecated
     public static final String METADATA_DELETE_AFTER_COMMIT = "metadata_delete_after_commit";
+    /**
+     * Please use  {@link TableProperties#METRICS_MAX_INFERRED_COLUMN_DEFAULTS}
+     */
+    @Deprecated
     public static final String METRICS_MAX_INFERRED_COLUMN = "metrics_max_inferred_column";
     public static final String TARGET_SPLIT_SIZE = TableProperties.SPLIT_SIZE;
-    private static final String DEFAULT_FORMAT_VERSION = "2";
+
+    private static final String DEPRECATION_WARNING_MESSAGE = "The table property \"%s\" has been " +
+            "deprecated. Please migrate to using \"%s\". This will become an error in future versions";
 
     private final List<PropertyMetadata<?>> tableProperties;
     private final List<PropertyMetadata<?>> columnProperties;
 
+    @VisibleForTesting
+    protected static final BiMap<String, String> DEPRECATED_PROPERTIES = ImmutableBiMap.<String, String>builder()
```

Comment:
I don't see where this is directly used outside of IcebergTableProperties.java. Why do you mark it VisibleForTesting and protected?

Reply:
initially it was meant to be used for tests but I went a different route. I updated the visibility